locator/network_topology_strategy.cc

```diff
@@ -554,6 +556,35 @@ tablet_replica_set network_topology_strategy::drop_tablets_in_dc(schema_ptr s, c
     return filtered;
 }
 
+void network_topology_strategy::sanity_check_read_replicas(const effective_replication_map& erm,
+                                                           const inet_address_vector_replica_set& read_replicas) const {
+    const auto& topology = erm.get_topology();
+
+    struct rf_node_count {
+        size_t replication_factor{0};
+        size_t node_count{0};
+    };
+
+    boost::unordered_flat_map<sstring, rf_node_count> data_centers_replication_factor;
+    std::ranges::for_each(read_replicas, [&data_centers_replication_factor, &topology, this](const auto& node) {
+        auto res = data_centers_replication_factor.emplace(topology.get_datacenter(node), rf_node_count{0, 0});
+        if (res.second) {
+            // For new item add replication factor.
+            res.first->second.replication_factor = get_replication_factor(res.first->first);
+        }
+        ++res.first->second.node_count;
+    });
+
+    std::ranges::for_each(data_centers_replication_factor, [](const auto& item) {
+        if (item.second.replication_factor < item.second.node_count) {
```

Comment:
So we ignore the CL of the read and simply check that DC replication matches the amount of replicas in the DC even though the read may be non LOCAL and eventually succeed. I guess this is fine if the goal is simply find a discrepancy between different subsystems in a debug build.